### PR TITLE
Migrate to WebClient

### PR DESCRIPTION
**Pull request by ghost**
RestTemplate is getting deprecated in favor of WebClient, according to the documentation. Therefore, it's better to use the modern WebClient.  Based on work done in PR #26.

### CODE REVIEW

**Comment by gabernard** [comment]
@alvasw Could you please review this PR? Thank you.

File: src/main/java/bisq/price/mining/providers/MempoolFeeRateProvider.java [hunk 1]
```diff
@@ -106,16 +102,13 @@ private FeeRate getEstimatedFeeRate() {
...
-                    // See https://github.com/bisq-network/projects/issues/27
-                    .fromUriString("https://" + getMempoolApiHostname() + "/api/v1/fees/recommended")
-                    .build().toUri())
-                .build(),
-            new ParameterizedTypeReference<Map<String, Long>>() { }
-        ).getBody().entrySet();
+    private Map<String, Long> getFeeRatePredictions() {
+        return WebClient.create().get()
+                .uri("https://" + getMempoolApiHostname() + "/api/v1/fees/recommended")
+                .accept()
```
Comment (alvasw) [inline src/main/java/bisq/price/mining/providers/MempoolFeeRateProvider.java]: You didn't set the `MediaType` in `.accept()`. It should be `.accept(MediaType.APPLICATION_JSON)`.

File: src/main/java/bisq/price/spot/providers/CoinGecko.java [hunk 2]
```diff
@@ -90,14 +84,11 @@ public Set<ExchangeRate> doGet() {
...
-                        .get(UriComponentsBuilder
-                                .fromUriString(CoinGecko.GET_EXCHANGE_RATES_URL).build()
-                                .toUri())
-                        .build(),
-                new ParameterizedTypeReference<CoinGeckoMarketData>() {
-                }
-        ).getBody();
+        return WebClient.create().get()
+                .uri(CoinGecko.GET_EXCHANGE_RATES_URL)
+                .accept()
```
Comment (alvasw) [inline src/main/java/bisq/price/spot/providers/CoinGecko.java]: You didn't set the `MediaType` in `.accept()`. It should be `.accept(MediaType.APPLICATION_JSON)`.

**Review by alvasw — CHANGES_REQUESTED** [review]
Looks good!

**Review by alvasw — APPROVED** [review]
utACK

**Review by gabernard — APPROVED** [review]
utACK

**Pull request merged**